beetsplug/similarity.py

```diff
@@ -0,0 +1,294 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2016, Rafael Bodill http://github.com/rafi
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+from __future__ import division, absolute_import, print_function
+
+import pylast
+from beets import ui
+from beets import config
+from beets import plugins
+from beets.dbcore import types
+
+import networkx as nx
+import matplotlib.pyplot as plt
+import os.path
+
+_artistsOwned = list()
+_artistsForeign = list()
+_relations = list()

```

Comment:
Do these need to be global variables, or would it work for them to be fields on the plugin class?